tests/test_builders/test_build_linkcheck.py

```diff
@@ -945,13 +954,15 @@ def test_connection_contention(get_adapter, app, capsys):
 
     # Place a workload into the linkcheck queue
     link_count = 10
-    rqueue, wqueue = Queue(), Queue()
+    wqueue: Queue[CheckRequest] = Queue()
+    rqueue: Queue[CheckResult] = Queue()
     for _ in range(link_count):
         wqueue.put(CheckRequest(0, Hyperlink("http://localhost:7777", "test", "test.rst", 1)))
 
     # Create parallel consumer threads
     with http_server(make_redirect_handler(support_head=True)):
-        begin, checked = time.time(), []
+        begin = time.time()
+        checked: list[CheckResult] = []
```

Comment:
If you want to keep it inline:

```suggestion
        begin, checked = time.time(), []  # type: (float, list[CheckResult])
```

Reply:
Nice to know that kind of typehint is possible; I think one-per-line is probably simpler here though.  And that `time` stuff could be worth revisiting later, so separating the line has some benefit.

Reply:
It's a legacy way to do it (the 2.7 way to do when `x: type` was not available). Note that there is an issue with this construction with mypy (it's an issue I discovered as well and patched but it's not merged yet: https://github.com/python/mypy/issues/16649).